crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1160,6 +1167,23 @@ impl<'db> Type<'db> {
                 false
             }
 
+            (Type::NominalInstance(_) | Type::ProtocolInstance(_), Type::Callable(_)) => {
+                let call_symbol = self.member(db, "__call__").symbol;
+                match call_symbol {
+                    Symbol::Type(Type::BoundMethod(call_function), _) => call_function
+                        .into_callable_type(db)
+                        .is_subtype_of(db, target),
+                    _ => false,
+                }
+            }
+            (Type::ProtocolInstance(left), Type::ProtocolInstance(right)) => {
+                left.is_subtype_of(db, right)
+            }
+            // A protocol instance can never be a subtype of a nominal type, with the *sole* exception of `object`.
+            // TODO: `Callable` types are also structural types.
```

Comment:
It's not totally clear to me what change this TODO is suggesting, since it seems like Protocol being a subtype of Callable is already handled two arms above?

Reply:
Oops, yes -- good catch! I originally planned to defer this to a followup PR, but mypy_primer highlighted several places where the PR was _introducing_ false positives due to protocols never being considered subtypes of Callable types. So I ended up just implementing it as part of this PR, since it's not too hard to do anyway. I'll get rid of the TODO.